<launch>
    <remap to="/camera_array/cam0/image_raw" from="/image_sender_0"/>

    <arg name="viz_output_path" default="/home/yang/video" />

    <group ns="vehicle">

        <node name="run_3cam_concat" pkg="mkz_intel" type="sync_3cams.py" output="screen" args="$(arg viz_output_path)"> </node>

        <node name="record_visualizations" pkg="image_view" type="video_recorder" output="screen">
            <remap to="sync_3cam" from="image"/>
            <param name="filename" value="$(arg viz_output_path).avi" />
            <param name="codec" value="X264" />
        </node>

        <node name="record_visualizations_enable" pkg="image_view" type="video_recorder" output="screen">
            <remap to="sync_3cam_enable" from="image"/>
            <param name="filename" value="$(arg viz_output_path)_enable.avi" />
            <param name="codec" value="X264" />
        </node>


        <node name="record_visualizations_info" pkg="image_view" type="video_recorder" output="screen">
            <remap to="/vis_continuous_full" from="image"/>
            <param name="filename" value="$(arg viz_output_path)_info.avi" />
            <param name="codec" value="X264" />
        </node>

        <node name="record_visualizations_info_enabled" pkg="image_view" type="video_recorder" output="screen">
            <remap to="/vis_continuous_full_enabled" from="image"/>
            <param name="filename" value="$(arg viz_output_path)_info_enable.avi" />
            <param name="codec" value="X264" />
        </node>

    </group>


</launch>